mpqp/tools/maths.py

```diff
@@ -62,15 +62,15 @@ def matrix_eq(lhs: Matrix, rhs: Matrix, atol: float = atol, rtol: float = rtol)
     Returns:
         ``True`` if the two matrix are equal (according to the definition above).
     """
-    from sympy import Expr
 
     for elt in zip(np.ndarray.flatten(lhs), np.ndarray.flatten(rhs)):
-        if isinstance(elt[0], Expr) or isinstance(elt[1], Expr):
-            if elt[0] != elt[1]:
-                return False
-        else:
+        try:
             if abs(elt[0] - elt[1]) > (atol + rtol * abs(elt[1])):
                 return False
+        except TypeError:
```

Comment:
Is this a clean way to handle the case when elements are of type `Expr`? What if we have another third case where the type is different from `Expr`, are we sure that comparing with the if is always working ?

Reply:
We have a case where Expr.number is compared to an int, and the result was false:
0.00 == 0 → false.
So, first, we perform a comparison, and if it doesn't work, we compare the elements directly. It's basically the same thing that before.

Reply:
basically, what happens is that it tries to do type coercion, and it it fails it does the naive check, I think it's quite clean

Reply:
Ok, thank you for the clarification